Token Rotation Improvements (#366)
ietf-wg-gnap/gnap-core-protocol: draft-ietf-gnap-core-protocol.md

URL, that the AS issued the token to the given client instance, and that
the presented key is appropriate to the token.

Note that in many cases, the access token will have expired for regular use. To facilitate
token rotation, the AS SHOULD honor the rotation request of the expired access token
since it is likely that the client instance is attempting to

> refresh the expired token. To support this, the AS MAY apply different lifetimes for

Reviewer: actually, what we're saying here is that the lifetime for token management might be longer that for its use

> refresh the expired token. To support this, the AS MAY apply different lifetimes for
> the use of the token in management vs. its use at an RS. An AS MUST NOT

Author: ```suggestion
refresh the expired token. To support this, the AS MAY allow a longer lifetime for
token management compared to its use at an RS. An AS MUST NOT
```

> honor a rotation request for an access token that has been revoked or otherwise disabled.

Reviewer: what do we mean by disabled ?

Author: Generically, just "made unusable for any reason whatsoever". Revocation is the most concrete -- and permanent -- version of that.

the token management URL as valid, since the end result is still the token
not being usable.

> Note that some deployment decisions like self-contained stateless access tokens can make

Reviewer: isn't it a bit redundant with what is already said above ? Note that stateless access tokens can make proactive
revocation difficult within a system, see {{security-stateless-tokens}}.

Author: I believe you're correct, this paragraph can be dropped.

> Note that some deployment decisions like self-contained stateless access tokens can make
> propagating revocation difficult, see {{security-stateless-tokens}}.

Author: ```suggestion
```

discussion of which are outside the scope of this specification.

However, the use of such self-contained access tokens has an effect on the ability of the AS to
provide certain functionality defined within this specification. Specifically, since the access
token is self-contained, it is difficult or impossible for an AS to signal to all RS's within an
ecosystem when a specific access token has been revoked. Therefore, an AS in such an ecosystem
should probably not offer token revocation functionality to client instances, since the client
instance's calls to such an endpoint is effectively meaningless. However, a client instance calling
the token revocation function will also throw out its copy of the token, so such a placebo endpoint
might not be completely meaningless. Token rotation similarly difficult because the AS has to
revoke the old access token after a rotation call has been made. If the access tokens are
completely self-contained and non-revocable, this means that there will be a period of time during
which both the old and new access tokens are valid and usable, which is an increased security risk
for the environment.

> These problems can be mitigated by keeping the validity time windows of self-contained access tokens

Reviewer: probably we should say that the mitigations should be applied for specific use cases, that aren't too sensitive
